core/src/main/scala/kafka/network/RequestChannel.scala

```diff
@@ -69,8 +69,16 @@ object RequestChannel extends Logging {
       } else
         null
     val body: AbstractRequest =
-      if (requestObj == null)
-        AbstractRequest.getRequest(header.apiKey, header.apiVersion, buffer)
+      if (requestObj == null) {
+        try {
+          // If the request version is higher than supported version, ArrayIndexOutOfBoundaryException might be thrown,
+          // we ignore it here and let KafkaApis to handle it.
```

Comment:
This doesn't look great, in my opinion. Can't we change `getRequest` to throw a more appropriate exception?